### PR TITLE
[testing] Switch to junit5

### PR DESCRIPTION
**Pull request by restingbull**
Fixes tests for modern jvms.

### CODE REVIEW

File: kotlin/internal/jvm/jvm.bzl [hunk 1]
```diff
@@ -344,6 +344,10 @@ Setup a simple kotlin_test.
             default = Label("@bazel_tools//tools/jdk:TestRunner_deploy.jar"),
             allow_files = True,
         ),
+        "use_testrunner": attr.bool(
```
Comment (Bencodes) [inline kotlin/internal/jvm/jvm.bzl:347]: What exactly is this flag solving for and when would you use it externally vs. setting it to `False`?

File: src/main/starlark/core/repositories/setup.bzl [hunk 2]
```diff
@@ -56,11 +56,17 @@ def kt_configure():
...
+            "org.junit.jupiter:junit-jupiter-api:5.4.2",
+            "org.junit.jupiter:junit-jupiter-params:5.4.2",
+            "org.junit.jupiter:junit-jupiter-engine:5.4.2",
+            "org.junit.platform:junit-platform-console:1.4.2",
         ],
         repositories = [
             "https://maven-central.storage.googleapis.com/repos/central/data/",
             "https://maven.google.com",
             "https://repo1.maven.org/maven2",
+            "https://jcenter.bintray.com",
```
Comment (Bencodes) [inline src/main/starlark/core/repositories/setup.bzl:69]: I believe jcenter is being shut down, best to not depend on it.

File: kotlin/internal/jvm/jvm.bzl (hunk 1 above)
Comment (restingbull) [reply kotlin/internal/jvm/jvm.bzl:347]: It matches [java_test.use_testrunner](https://bazel.build/reference/be/java#java_test.use_testrunner). It allows customizing the test execution environment. In this case, we replace the bazel test runner with JUnit5.

File: src/main/starlark/core/repositories/setup.bzl (hunk 2 above)
Comment (restingbull) [reply src/main/starlark/core/repositories/setup.bzl]: Removed.

**Review by Bencodes — APPROVED** [review]

**Pull request merged**